javascript/MaterialXView/source/helper.js

```diff
@@ -117,18 +117,65 @@ function toThreeUniform(type, value, name, uniforms, textureLoader, searchPath,
         case 'filename':
             if (value)
             {
-                let fullPath = searchPath + IMAGE_PATH_SEPARATOR + value;
-                const texture = textureLoader.load(fullPath);
-                // Set address & filtering mode
-                if (texture)
-                    setTextureParameters(texture, name, uniforms, flipY);
-                outValue = texture;
+                // Cache / reuse texture to avoid reload overhead.
+                // Note: that data blobs and embedded data textures are not cached as they are transient data.
+                let checkCache = true;
+                let texturePath = searchPath + IMAGE_PATH_SEPARATOR + value;
+                if (value.startsWith('blob:'))   
```

Comment:
Blobs, embedded data and http references are no relative to a search path. This fixes it so users can use these texture URI specifications in their files.